src/imagelightbox.js

```diff
@@ -202,13 +202,18 @@
                 });
             },
             captionOn = function () {
-                var description = $(target).find('img').attr('alt');
+                var description;//
+                if ($(target).find('img').length) {
+                    description = $(target).find('img').attr('alt');
+                } else {
+                    description = $(target).data("imagelightbox-caption");
+}
```

Comment:
I would go with the first approach (if you meant to rephrase my idea). Using the alt-attribute for other stuff (like thumbnail overlay) isnt its purpose. And the user can already decide by using data-attributes.